plugins/terminal/ios.py

```diff
@@ -92,7 +116,11 @@ def on_become(self, passwd=None):
                 to_bytes(json.dumps(cmd), errors="surrogate_or_strict")
             )
             prompt = self._get_prompt()
-            if prompt is None or not prompt.endswith(b"#"):
+            if (
+                prompt is None
+                or not prompt.endswith(b"#")
+                or self.get_privilege_level() != 15
+            ):
                 raise AnsibleConnectionFailure(
                     "failed to elevate privilege to enable mode still at prompt [%s]"
                     % prompt
```

Comment:
Do you want to reference the current privilege level here also?

Reply:
That might be redundant.